packages/connect-web/src/impl/core-in-popup.ts

```diff
@@ -130,26 +130,26 @@ export class CoreInPopup implements ConnectFactoryDependencies {
             await this._popupManager.request();
         }
 
-        await this._popupManager.channel.init();
-
-        if (this._settings.env === 'webextension') {
-            // In webextension we init based on the popup promise
-            // In core this is handled in the popup manager
-            await this._popupManager.popupPromise?.promise;
-
-            this._popupManager.channel.postMessage({
-                type: POPUP.INIT,
-                payload: {
-                    settings: this._settings,
-                    useCore: true,
-                },
-            });
-        }
+        try {
+            await this._popupManager.channel.init();
+
+            if (this._settings.env === 'webextension') {
+                // In webextension we init based on the popup promise
+                // In core this is handled in the popup manager
+                await this._popupManager.popupPromise?.promise;
+
+                this._popupManager.channel.postMessage({
+                    type: POPUP.INIT,
+                    payload: {
+                        settings: this._settings,
+                        useCore: true,
+                    },
+                });
+            }
 
-        await this._popupManager.handshakePromise?.promise;
+            await this._popupManager.handshakePromise?.promise;
 
-        // post message to core in popup
-        try {
+            // post message to core in popup
```

Comment:
what is the reason for this change?

Reply:
We want to catch all errors and return `createErrorMessage`